lib/feeder_importer.rb

```diff
@@ -1,64 +1,153 @@
+  belongs_to :episode
+  enum status: status_values
+end
+
+class Content < MediaResource; end
+class Enclosure < MediaResource; end
+# Feeder models end
 
 class FeederImporter
-  include PRXAccess
   include Announce::Publisher
+  include PRXAccess
 
-  attr_accessor :account_id, :user_id, :feeder_podcast_url
-  attr_accessor :podcast, :series, :template, :distribution
+  attr_accessor :account_id, :user_id, :podcast_id, :set_episode_urls
+  attr_accessor :podcast, :series, :template, :distribution, :stories
+
+  def debug
+    TRUE
+  end
 
-  def initialize(account_id, user_id, feeder_podcast_url)
+  def initialize(account_id, user_id, podcast_id, set_episode_urls = false)
     self.account_id = account_id
     self.user_id = user_id
-    self.feeder_podcast_url = feeder_podcast_url
+    self.podcast_id = podcast_id
+    self.stories = []
+    self.set_episode_urls = set_episode_urls
   end
 
   def import
     retrieve_podcast
     create_series
+    create_stories
+    update_podcast
   end
 
-  def create_series(pcast = podcast)
+  def retrieve_podcast
+    self.podcast = Podcast.find(podcast_id)
+  end
+
+  def create_series
     attrs = {
       app_version: PRX::APP_VERSION,
       account_id: account_id,
       creator_id: user_id,
-      title: pcast.title,
-      short_description: pcast.attributes[:subtitle],
-      description_html: pcast.attributes[:description]
+      title: podcast.title,
+      short_description: podcast.subtitle,
+      description_html: podcast.description
     }
     self.series = Series.create!(attrs)
 
-    # Add images to the series
-    if pcast.attributes['itunes_image'] && pcast.itunes_image['url']
-      image = series.images.create!(
-        upload: clean_string(pcast.itunes_image['url']),
-        purpose: Image::PROFILE
-      )
-      announce_image(image)
-    end
+    podcast.itunes_images.each { |i| create_series_image(i, Image::PROFILE) }
+    podcast.feed_images.each { |i| create_series_image(i, Image::THUMBNAIL) }
 
-    if pcast.attributes['feed_image'] && pcast.feed_image['url']
-      image = series.images.create!(
-        upload: clean_string(pcast.feed_image['url']),
-        purpose: Image::THUMBNAIL
-      )
-      announce_image(image)
-    end
+    # all the imports we plan to do from feeder -> cms have a single segment
+    num_segments = 1
 
-    # Add the template and a single file template
     self.template = series.audio_version_templates.create!(
-      label: 'Podcast Audio',
+      label: "Podcast Audio #{num_segments} #{'segment'.pluralize(num_segments)}",
+      segment_count: num_segments,
       promos: false,
       length_minimum: 0,
       length_maximum: 0
     )
 
-    num_segments = 1
-    episode = pcast.episodes.first
-    if episode.attributes['media']
-      num_segments = [episode.media.count, num_segments].max
+    episode = podcast.episodes.first
+    if episode.media_resources
+      num_segments = [episode.media_resources.count, num_segments].max
```

Comment:
There should always be 1, as the comment indicates, but I worry that this code might be used for other cases eventually, and wrote the code to handle more if necessary.

Feeder definitely allows more than one file, so I thought it safer to handle what could be the case in feeder, especially since it is a pretty small amount of added code/complexity.

The code to actually add the audio files from feeder to an episode likewise doesn't only get `first`, it looks through all of them, b/c that is what the feeder model supports.